project-fob/Controllers/HostController.cs

```diff
@@ -23,15 +24,35 @@ public ActionResult Index()
         {
             return View();
         }
+
+        public ActionResult QrCode()
+        {
+            var gotValue = HttpContext.Session.TryGetValue("meetingid", out var meetingIdValue);
+            string meetingIdString = System.Text.Encoding.ASCII.GetString(meetingIdValue);
+
+            string baseUrl = Request.GetDisplayUrl();
+            string[] split = baseUrl.Split('/');
+            StringBuilder sb = new StringBuilder();
+
+            for (int i = 0; i < split.Length - 2; i++)
+            {
+                sb.Append(split[i]);
+            }
+            Meeting meet = db.Meeting.Single(x => x.MeetingId.Equals(meetingIdString));
```

Comment:
I think it's worth extracting the URL building bit to its own method to make this a bit clearer (i.e the part in lines 33-41)